sympy/combinatorics/perm_groups.py

```diff
@@ -3815,7 +3815,301 @@ def _sylow_reduce(mu, nu):
+                        i_gamma = delta.index(gamma)
+                        d = gamma^g
+                        i_d = delta.index(d)
+                        if p[i_d] < 0:
+                            u[i_d] = u[i_gamma]*g
+                            p[i_d] = sigma
+                            blocks[i_d] = rep
+                            newb.append(d)
+                        else:
+                            # B[rho] is not a block
+                            s = u[i_gamma]*g*u[i_d]**(-1)
+                            return False, s
+
+                    B.append(newb)
+                else:
+                    for h in B[rho][1:]:
+                        if not h^g in B[sigma]:
+                            # B[rho] is not a block
+                            s = u[delta.index(beta)]*g*u[i_d]**(-1)
+                            return False, s
+            rho += 1
+
+        return True, blocks
+
+    def _verify(H, K, phi, z, alpha):
+        '''
+        Return a list of relators `rels` in generators `gens_h` that
+        are mapped to `H.generators` by `phi` so that given a finite
+        presentation <gens_k | rels_k> of `K` on a subset of `gens_h`
+        <gens_h | rels_k + rels> is a finite presentation of `H`.
+
+        `H` should be generated by the union of `K.generators` and `z`
+        (a single generator), and `H.stabilizer(alpha) == K`; `phi` is a
+        canonical injection from a free group into a permutation group
+        containing `H`.
+
+        The algorithm is described in [1], Chapter 6.
+
+        Example
+        =======
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> from sympy.combinatorics import Permutation
+        >>> from sympy.combinatorics.homomorphisms import homomorphism
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> from sympy.combinatorics.fp_groups import FpGroup
+
+        >>> H = PermutationGroup(Permutation(0, 2), Permutation (1, 5))
+        >>> K = PermutationGroup(Permutation(5)(0, 2))
+        >>> F = free_group("x_0 x_1")[0]
+        >>> gens = F.generators
+        >>> phi = homomorphism(F, H, F.generators, H.generators)
+        >>> rels_k = [gens[0]**2] # relators for presentation of K
+        >>> z= Permutation(1, 5)
+        >>> check, rels_h = H._verify(K, phi, z, 1)
+        >>> check
+        True
+        >>> rels = rels_k + rels_h
+        >>> G = FpGroup(F, rels) # presentation of H
+        >>> G.order() == H.order()
+        True
+
+        See also
+        ========
+        strong_presentation, presentation, stabilizer
+
+        '''
+
+        orbit = H.orbit(alpha)
+        beta = alpha^(z**-1)
+
+        K_beta = K.stabilizer(beta)
+
+        # orbit representatives of K_beta
+        gammas = [alpha, beta]
+        orbits = list(uniq(K_beta.orbit(o) for o in orbit))
+        orbit_reps = [list(orb)[0] for orb in orbits]
+        for rep in orbit_reps:
```

Comment:
Is it clear that `alpha` would be the first element, `list(orb)[0]`, of the K_beta-orbit `orb` of `alpha`? If not, then we should probably remove that orbit from `orbits` (and also the orbit of `beta`).

Reply:
`K_beta` stabilizes `beta` (by definition) and `alpha` (because `K == H.stabilizer(alpha)`) so they are the only things in their orbits. Though excluding them when `orbits` are computed would probably be a little faster.

Reply:
Of course. I forgot that `K_beta` also stabilizes `alpha`. This should be OK then.